src/pages/homepage/Homepage.jsx

```diff
@@ -11,17 +11,42 @@ export default function Homepage() {
   const [pageNumber, setPageNumber] = useState(1);
   const [limitNumber, setLimitNumber] = useState(5);
   const navigate = useNavigate();
-
   const { topic } = useParams();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sortByQuery = searchParams.get("sort_by");
+  const orderByQuery = searchParams.get("order_by");
+  const location = useLocation();
+
+  function stringify(object) {
+    let finalString = "?";
+    for (let key in object) {
+      finalString += `${key}=${object[key]}&`;
+    }
+    finalString = finalString.slice(0, -1);
+    return finalString;
+  }
+
+  function objectify(queryString) {
+    if (queryString === "") {
+      return {};
+    }
+    const object = {};
+    const queryArray = queryString.replace("?", "").split("&");
+    queryArray.forEach((keyValuePair) => {
+      const keyValueArray = keyValuePair.split("=");
+      object[keyValueArray[0]] = keyValueArray[1];
+    });
+    return object;
+  }
```

Comment:
Well done for getting the page URL to update to reflect the sort_by and order, it's a tricky one!